core/config/project_settings.cpp

```diff
@@ -723,9 +723,9 @@ Error ProjectSettings::_load_settings_text(const String &p_path) {
 	Error err;
 	Ref<FileAccess> f = FileAccess::open(p_path, FileAccess::READ, &err);
 
-	if (f.is_null()) {
-		// FIXME: Above 'err' error code is ERR_FILE_CANT_OPEN if the file is missing
-		// This needs to be streamlined if we want decent error reporting
+	if (err != OK) {
+		return err;
+	} else if (f.is_null()) {
 		return ERR_FILE_NOT_FOUND;
 	}
```

Comment:
It's impossible to have both OK and null, so the second check is redundant.

Reply:
https://github.com/godotengine/godot/pull/81891#issuecomment-1725337092  is it redudant ?

Reply:
Well that comment seems to come with the same concerns as https://github.com/godotengine/godot/pull/81958#issuecomment-1732669130
My point is that if `err` is `OK`, the FileAccess **will** exist (unless it's bugged). And again, `_load_settings_binary()` doesn't do any null check either.